bin/adhoc-scripts/updatePileupObjects.py

```diff
@@ -147,6 +158,65 @@ def writePileupDocs(mspileupUrl, puDocs, handler, logger):
                 logger.critical(msg)
 
 
+def addTransitionRecords(handler, url, userDN, logger, dryrun):
+    """
+    Helper function to add and update in place transition record for existing MSPileup records
+    :param handler: HTTP handler to use
+    :param url: url to use (string)
+    :param userDN: string representing user DN
+    :param logger: logger to use
+    :param dryrun: option to run dry-run mode (boolean)
+    :return: nothing
+    """
+    puDocs = getPileupDocs(url, handler, logger)
+    logger.info("Found %d documents in MSPileup", len(puDocs))
+    for rec in puDocs:
+        if not rec.get("transition"):
+            tranRecord = {'containerFraction': 1.0,
+                          'customDID': rec['pileupName'],
+                          'updateTime': gmtimeSeconds(),
+                          'DN': userDN}
+            rec['transition'] = [tranRecord]
+            logger.info("New pileup document is: %s", rec)
+
+    if dryrun:
+        logger.info("documents are not written due to --dry-run option\n")
+    else:
+        # finally, update the pileup documents in the database
+        writePileupDocs(url, puDocs, handler, logger)
+        logger.info("documents are uploaded to MSPileup\n")
+
+
+def updatePileupRecords(handler, url, fin, logger, dryrun):
+    """
+    helper function to update pileup records
+    :param handler: HTTP handler to use
+    :param url: url to use (string)
+    :param fin: input file name to read from (string)
+    :param logger: logger object to use
+    :param dryrun: option to run dry-run mode (boolean)
+    :return: nothing
+    """
+    with open(fin, 'r', encoding='utf-8') as istream:
+        puOverride = json.load(istream)
+    logger.info("Pileup override file: %s has the following content: %s", fin, puOverride)
+
+    puDocs = getPileupDocs(url, handler, logger)
+    logger.info("Found %d documents in MSPileup", len(puDocs))
+
+    # update the documents with the override content
+    for doc in puDocs:
+        doc.update(puOverride)
+        logger.info("New pileup document is: %s", doc)
+
+    if dryrun:
+        logger.info("documents are not written due to --dry-run option\n")
```

Comment:
If it's dry-run, should we print the documents such that they can be inspected before actually writing them back into the database?

Reply:
ok, I added printout of docs.